iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,341 @@
+        {
+            return UpdateAsync(configuration, false, cancellationToken);
+        }
+
+        /// <summary>
+        /// Update the mutable fields of the configuration registration
+        /// </summary>
+        /// <param name="configuration">The configuration object with updated fields.</param>
+        /// <param name="forceUpdate">Forces the configuration object to be replaced even if it was updated since it was retrieved last time.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object with updated ETags.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task<Configuration> UpdateAsync(Configuration configuration, bool forceUpdate, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Updating configuration: {configuration?.Id} - Force update: {forceUpdate}", nameof(UpdateAsync));
+
+            try
+            {
+                if (string.IsNullOrWhiteSpace(configuration.ETag) && !forceUpdate)
+                {
+                    throw new ArgumentException(ApiResources.ETagNotSetWhileUpdatingConfiguration);
+                }
+
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Put, GetConfigurationRequestUri(configuration.Id), _credentialProvider, configuration);
+                HttpMessageHelper2.InsertEtag(request, configuration.ETag);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response);
+                return await HttpMessageHelper2.DeserializeResponse<Configuration>(response, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(UpdateAsync)} threw an exception: {ex}", nameof(UpdateAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Updating configuration: {configuration?.Id} - Force update: {forceUpdate}", nameof(UpdateAsync));
+            }
+        }
+
+        /// <summary>
+        /// Deletes a previously registered device from the system.
+        /// </summary>
+        /// <param name="configurationId">The id of the configurationId being deleted.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task RemoveAsync(string configurationId, CancellationToken cancellationToken = default)
```

Comment:
Azure SDK team guidance dictates that we use the verb "Delete" instead of "Remove" for operations like these.
```suggestion
        public async virtual Task DeleteAsync(string configurationId, CancellationToken cancellationToken = default)
```

Reply:
https://azure.github.io/azure-sdk/dotnet_introduction.html